tast_three.py

```diff
@@ -0,0 +1,49 @@
+# Создать форму для регистрации пользователей на сайте.
+# Форма должна со держать поля "Имя", "Фамилия", "Email", "Пароль" и кнопку "Зарегистрироваться".
+# При отправке формы данные должны сохраняться в базе данных, а пароль должен быть зашифрован.
+
+from flask import Flask, render_template, redirect, request, url_for
+from hashlib import sha256
+from flask_wtf.csrf import CSRFProtect
+from form import RegisterForm
+from models import db, User
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///forms.db'
+app.config['SECRET_KEY'] = b'32hk4g432543jk3l5hkh53h24223ji56igj4n4d'
+csrf = CSRFProtect(app)
+db.init_app(app)
+
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    form = RegisterForm()
+    if form.validate_on_submit():
+        user = User(first_name=form.first_name.data,
+                    last_name=form.last_name.data,
+                    email=form.email.data,
+                    password=sha256(form.password.data.encode(encoding='utf-8')).hexdigest())
+        db.session.add(user)
+        db.session.commit()
+        return redirect(url_for('index'))
+    return render_template('/index.html', form=form)
+
```

Comment:
Порой записи в БД не происходит и не понятно в чем ошибка. Все дело в том, что форма не проходит валидацию и на этот счет не предусмотрены никакие действия чтобы об этом сообщить. Можно сделать хотя бы так.
```suggestion
    else:
        for obj in form:
            for error in obj.errors:
                flash(error)
    return render_template('/index.html', form=form)
```